tools.py

```diff
@@ -344,7 +344,7 @@ def cancel(self):
 
 def getAllTools():
     this_mod = __import__(__name__)
-    all = [val for val in this_mod.__dict__.values() if isinstance(val, type)]
+    all = [val for val in list(this_mod.__dict__.values()) if isinstance(val, type)]
```

Comment:
I guess the `list()` operator was not necessary. Moreover `all` is a built in function that changed from python2 to python3. It might be better to change `all` variable name to something different so that it does not shadow the builtin function